### ISSUE TITLE
Epic: Blazor developers can target and deploy their applications to desktop platforms, allowing them to run natively

### ISSUE THREAD

**Issue by maddyleger1** [dotnet/aspnetcore#27217]
### Summary

In .NET 6 we want to extend Blazor's target scenarios to include native desktop applications via web-based rendering.

Note: This is a proposed experience for .NET 6 and not yet committed.

`Todo: add linked user stories`

**Comment by RChrisCoble** [comment]
Are other platforms in play here as well?  Like MBB's usage of iOS and Android to complete this story?  Or maybe the inclusion of MBB into ASP.Net brings those and this epic fills the gap?

**Comment by Eilon** [comment]
I'm in the loops, One .NET!

**Comment by RChrisCoble** [comment]
@Eilon !!!  I for one welcome our One .Net overlords.  

For me, there's one big piece which brings Blazor together in a powerhouse .Net 6 release, but it's over on the Runtime team.  @danroth27 can you throw your weight around to get this slotted between now and November?

[.NET WebAssembly support of task based parallelism. #40619](https://github.com/dotnet/runtime/issues/40619)

**Comment by wakuflair** [comment]
Is it something with MAUI?

**Comment by kristof12345** [comment]
Hi! 
Any update on this? Can we expect this feature in .NET 6?
Thank you.

**Comment by Eilon** [comment]
Hi @kristof12345 yes this is part of the plans for .NET 6!

**Comment by Pilchie** [comment]
:eyes:

**Comment by mrlife** [comment]
With MAUI introducing support for the Android and iOS platforms, and since Blazor is built on top of MAUI, will Blazor support those platforms as well?

**Comment by Eilon** [comment]
@mrlife the main goal for .NET 6 is to support the Windows and macOS desktop platforms, but we will surely have code running on Android and iOS as well, but it might not be fully supported at that time.

**Comment by leonardder** [comment]
I hope that accessibility for impaired people (like blindness) will be accounted for so apps like these will be accessible from the start, or at least the examples will be.

**Comment by kmanev073** [comment]
Is this available in the .NET 6 Preview 1? Thanks in advance.

**Comment by todosrc** [comment]
I want Blazor Desktop to also support automatically update themselves. like electron-updater.

Idea :

https://github.com/ElectronNET/Electron.NET/issues/259
https://github.com/ElectronNET/Electron.NET/blob/master/ElectronNET.API/AutoUpdater.cs

https://github.com/NetSparkleUpdater/NetSparkle
https://github.com/Squirrel

cheers>>

**Comment by Eilon** [comment]
@leonardder accessibility is certainly important for all apps so I'm adding that to the checklist of things to look into.

**Comment by Eilon** [comment]
@kmanev073 said:

> Is this available in the .NET 6 Preview 1? Thanks in advance.

It is not in Preview 1. We plan to have this in no later than Preview 4.

**Comment by Eilon** [comment]
@todosrc said:

> I want Blazor Desktop to also support automatically update themselves. like electron-updater.

Installation/deployment/update support is an interesting problem, and we are looking at various solutions to this, including using some of the ones you mention!

**Comment by birbilis** [comment]
Remember ClickOnce deployment?

**Comment by robertmclaws** [comment]
So, something to consider while you folks are looking into this.

I'm building BurnRate.io, and we're in the process of moving a large OData/TypeScript/Webpack app to Blazor. When we add support for installing the app to the Desktop, we're still planning on shipping our Blazor WebAssembly version to the web, so people can access it where- and however they want. Up to this point, we had planned on doing that through Electron.NET.

However, Electron's update infrastructure is less than awesome, and requires a lot to build multiplatform update binaries. This should be wholly unnecessary with Blazor.

It would be great if the Blazor Desktop version could just download a manifest of MD5 hashes for the app files, and if any of them differ from what is installed, the app downloads them from the server-deployed version. That way, it's seamless for everyone, I know my customers will always be running the latest version, and you don't have to worry about trying to package MSI files or anything like that.

If security is an issue, then the files could all be digitally signed on the server, to reduce the risk of tampering.

I hope that real-world situation helps with your planning. We are a Microsoft for Startups / Microsoft Partner company, so if you have a TAP that would let us test this before the preview, would love to participate. Thanks so much!

**Comment by Webreaper** [comment]
This would be totally awesome. My app (http://github.com/webreaper/damselfly) bundles the installers/zip for the desktop electron wrapper into the Docker images, and I've used some javascript interop between Electron and the server to determine if the current container matches the server that's providing its content. If not, the user is notified and can optionally download the DMG/zip/AppImage file from the server and upgrade. I'll probably implement one-click download/install at some point in the future. But it would be great if Blazor could provide all the scaffolding to do this automatically without me having to write the versioning management and server/desktop JS interop by hand.

**Comment by rogihee** [comment]
I think if Blazor Desktop is built on top of MAUI with WinUI3 which requires Win10 1809, then MSIX might be a good option as well.

**Comment by JohnGalt1717** [comment]
@rogihee My perfect world is somewhere very close to that too. I'd like to see WinUI 3 across all of MAUI including Web and all other platforms and that Blazor Desktop would just be a markup syntax for MAUI...

**Comment by Lucifer-Morning-Star** [comment]
> @mrlife the main goal for .NET 6 is to support the Windows and macOS desktop platforms, but we will surely have code running on Android and iOS as well, but it might not be fully supported at that time.

So, would it not support Linux desktop loatforms?

**Comment by shashank-shekhar** [comment]
> 
> 
> @kmanev073 said:
> 
> > Is this available in the .NET 6 Preview 1? Thanks in advance.
> 
> It is not in Preview 1. We plan to have this in no later than Preview 4.

@Eilon is there a planned cadence for the preview releases?

**Comment by Eilon** [comment]
@shashank-shekhar - This is all part of .NET 6, so when the code is ready it will be in the next .NET 6 preview. We're hoping to get _something_ into .NET 6 Preview 3, but no later than .NET 6 Preview 4.

**Comment by Eilon** [comment]
@Lucifer-Morning-Star said:

> So, would it not support Linux desktop loatforms?

Linux is not currently planned, but with .NET MAUI (much like Xamarin.Forms) it's conceivable that it could be added in the future.

**Comment by tub5** [comment]
> @Eilon said:
> Linux is not currently planned, but with .NET MAUI (much like Xamarin.Forms) it's conceivable that it could be added in the future.

Damn, that's a deal breaker for us. Looks like we'll have to stick with Electron.NET for the foreseeable.

**Comment by Kryptos-FR** [comment]
@tub5 have a look at Avalonia UI or Uno.

**Comment by tub5** [comment]
@Kryptos-FR we looked at both previously but decided using Blazor with Electron.NET would suffice until the Blazor Desktop Applications came about. As our application is written using Blazor, when the ability is implemented, the migration should take a matter of days rather than weeks/months recreating the UI.

**Comment by Lucifer-Morning-Star** [comment]
> @Lucifer-Morning-Star said:
> 
> > So, would it not support Linux desktop platform?
> 
> Linux is not currently planned, but with .NET MAUI (much like Xamarin.Forms) it's conceivable that it could be added in the future.

So, 

> @Lucifer-Morning-Star said:
> 
> > So, would it not support Linux desktop patform?
> 
> Linux is not currently planned, but with .NET MAUI (much like Xamarin.Forms) it's conceivable that it could be added in the future.
So, Is it  possible give a way to make the Blazor applications to desktop platforms just like SteveSanderson's attempt [.](https://github.com/SteveSandersonMS/WebWindow) ,or make this lib have a  official support in .NET 6 ?It may be a easy way to support Linux desktop platforms in a short time.

**Comment by RChrisCoble** [comment]
I believe Linux was only descoped from .Net 6 during their planning.  I would imagine we'll see that re-scoped for .Net 7.  

You can see [on this page the Feature still mentions Linux](https://github.com/dotnet/aspnetcore/issues/27217), but they removed it from the actual feature [once you click on it.](https://github.com/dotnet/aspnetcore/issues/27234)

**Comment by Lucifer-Morning-Star** [comment]
> I believe Linux was only descoped from .Net 6 during their planning. I would imagine we'll see that re-scoped for .Net 7.
> 
> You can see [on this page the Feature still mentions Linux](https://github.com/dotnet/aspnetcore/issues/27217), but they removed it from the actual feature [once you click on it.](https://github.com/dotnet/aspnetcore/issues/27234)

This really is a bad news😒. Hope it would come in .NET 7.

**Comment by RChrisCoble** [comment]
> > I believe Linux was only descoped from .Net 6 during their planning. I would imagine we'll see that re-scoped for .Net 7.
> > You can see [on this page the Feature still mentions Linux](https://github.com/dotnet/aspnetcore/issues/27217), but they removed it from the actual feature [once you click on it.](https://github.com/dotnet/aspnetcore/issues/27234)
> 
> This really is a bad news😒. Hope it would come in .NET 7.

It's [being discussed here](https://github.com/jsuarezruiz/forms-gtk-progress/issues/31), weigh in on what you think is important.

**Comment by sinand99** [comment]
Trying to run web apps on desktop is plain stupid. If you want desktop app, build a native app, so simple.
You should focus your energy to make Blazor to work better on browsers.

**Comment by RChrisCoble** [comment]
> Trying to run web apps on desktop is plain stupid. If you want desktop app, build a native app, so simple.
> You should focus your energy to make Blazor to work better on browsers.

Depends on your use case.  Some of us would like to create an experience that can execute on Web, OnPrem, Mobile, etc. using the same code base and UI layer.  From that perspective Blazor Desktop is awesome from a "write once deploy everywhere" standpoint.  Don't try that with MAUI, or any other Windows UI technology ever created previously.

**Comment by thomaslevesque** [comment]
> Trying to run web apps on desktop is plain stupid. If you want desktop app, build a native app, so simple.
> You should focus your energy to make Blazor to work better on browsers.

I think you underestimate the value of having a single framework to build apps on the web and on many desktop platforms, with a single code base. Native apps are fine (and actually better in some ways, e.g. they typically have better performance), but you need to redevelop them for every platform you want to support, possibly in addition to the web app. Why do you think Electron is so successful, even though it creates horribly bloated apps?

**Comment by tub5** [comment]
> Trying to run web apps on desktop is plain stupid. If you want desktop app, build a native app, so simple.
> You should focus your energy to make Blazor to work better on browsers.

This is such a horrible take. .NET Core/5.0+ is designed to be multi-platform. The core functionality, i.e. the backend, compiles down to native. Why wouldn't you want to use a single way to create a user interface? What is the difference between using Blazor to generate this interface over XAML through Xamarin/WinForms/WPF?

The amount of issues through different implementations on different operating systems would dramatically reduce and it would slash the amount of development time enhancing and migrating features.

**Comment by sinand99** [comment]
> 
> 
> > Trying to run web apps on desktop is plain stupid. If you want desktop app, build a native app, so simple.
> > You should focus your energy to make Blazor to work better on browsers.
> 
> I think you underestimate the value of having a single framework to build apps on the web and on many desktop platforms, with a single code base. Native apps are fine (and actually better in some ways, e.g. they typically have better performance), but you need to redevelop them for every platform you want to support, possibly in addition to the web app. Why do you think Electron is so successful, even though it creates horribly bloated apps?

Yes, write-once-run-everywhere sounds wonderful for lazy developers but it should be done with a proper framework which compiles the code to its native controls on every platform and runs natively, not by creating some web controls and emulating them in a fake browser. I think that Microsoft is planning to do the latter when they say "web-based rendering". I just don't want Blazor to turn into Electron fiasco. It is like creating a windows app and packing it into a VM just to force it to work under Linux or MacOS (which has so little userbase compared to Windows btw).  It just creates bug-ridden montrosites and have horrible performance.

**Comment by sinand99** [comment]
> 
> 
> > Trying to run web apps on desktop is plain stupid. If you want desktop app, build a native app, so simple.
> > You should focus your energy to make Blazor to work better on browsers.
> 
> This is such a horrible take. .NET Core/5.0+ is designed to be multi-platform. The core functionality, i.e. the backend, compiles down to native. Why wouldn't you want to use a single way to create a user interface? What is the difference between using Blazor to generate this interface over XAML through Xamarin/WinForms/WPF?
> 
> The amount of issues through different implementations on different operating systems would dramatically reduce and it would slash the amount of development time enhancing and migrating features.

If you don't know the difference between Blazor and WinForms, you should educate yourself then. Blazor' goal is to replace WebForms, not "a single way to create a user interface". That is the MAUI's job. But there is a long time until that, since true multi-platfrom is still a dream imo. Your super multi-platfrom app will stop working when any of the OS you support implements a breaking change via a major update. You have to constantly maintain it to keep it working and behaving/looking the same on all platforms. As a software developer, you wouldn't wanna give your users a bloated crap, just to make it cross-platform.

**Comment by Webreaper** [comment]
Sinan, Blazor's job is not "to replace Webforms", there's much more to it than that. And all the problems you describe are solved by using browser/HTML based UIs, implicitly. Writing individual native UIs for every platform is pointless when it's possible to write a great UI that'll work on all platforms (including mobile) without even having to buy hardware for the other platforms.

I'm not sure what you're trying to suggest with the line you're going down - but to take the line you're taking on this issue seems somewhat strange to me. If you want to write individual native UIs for all platforms, there's nothing stopping you doing that, but for those of us who are happier with write-once, run anywhere, this Epic is *exactly* what we're looking for.

**Comment by tub5** [comment]
> If you don't know the difference between Blazor and WinForms, you should educate yourself then.

You're making yourself look like a fool. Stop being so rude.

> Blazor' goal is to replace WebForms, not "a single way to create a user interface"

WebForms died off a long time ago, I have no idea why you think Blazor is directly replacing it?
WinForms was a simple example of creating an application with a GUI, the same way you will be able to with Blazor Desktop Applications, not a literal comparison.

> That is the MAUI's job. But there is a long time until that, since true multi-platfrom is still a dream imo

That is one option. Why wouldn't you allows developers the option to choose which they would prefer. Electron has been around for a while and has become a big success. Doing the same for Blazor whilst improving efficiency and lowering the resources/package size is logical.

>  Your super multi-platfrom app will stop working when any of the OS you support implements a breaking change via a major update. You have to constantly maintain it to keep it working and behaving/looking the same on all platforms

And that is going to change with Xamarin/MAUI?

> As a software developer, you wouldn't wanna give your users a bloated crap, just to make it cross-platform.

Again, that is going to change with Xamarin/MAUI? Apps created for Android using Xamarin were found to actually run faster and were far lighter than those built using Android Studio.

**Comment by b-straub** [comment]
IMHO neither GitHub nor this Epic is the right place for this kind of discussion, this is still a free world and nobody prevents any developer in any way choosing the right platform for his needs, the right OS, the right smartphone or whatever. Please let’s keep this discussion what it was meant for and stop evangelism each other.

**Comment by Eilon** [comment]
Hi everyone, let's try to lower the temperature here just a bit... It's important to remain respectful of everyone's opinions even when we disagree.

**Comment by garrynewman** [comment]
Just want to throw my support in for this. I [made a Blazor desktop app](https://files.facepunch.com/garry/4876690e-0d1b-4058-9626-43e30c8537cf.png) and it was lots of fun and a really nice way to work.

**Comment by NickBehrends** [comment]
> @shashank-shekhar - This is all part of .NET 6, so when the code is ready it will be in the next .NET 6 preview. We're hoping to get _something_ into .NET 6 Preview 3, but no later than .NET 6 Preview 4.

@Eilon is this still planned for Preview 4? I see we got MAUI for windows in Preview 3 but no mention of blazor, wanted to check in as I don't see it on the [Preview 4 WIP Epic](https://github.com/dotnet/core/issues/6098).

**Comment by Eilon** [comment]
Oh yeah @NickBehrends you bet it's in! The .NET MAUI parts of BlazorWebView are all in the https://github.com/dotnet/maui repo, so it's not always easy to see the progress from other repos. The .NET MAUI BlazorWebView code is already in, and I'm working on the project templates now. All the previews ship at the essentially the same time (as far as I know), so you'll have it in your hands very soon!

**Comment by MichaelPeter** [comment]
> Oh yeah @NickBehrends you bet it's in! The .NET MAUI parts of BlazorWebView are all in the https://github.com/dotnet/maui repo, so it's not always easy to see the progress from other repos. The .NET MAUI BlazorWebView code is already in, and I'm working on the project templates now. All the previews ship at the essentially the same time (as far as I know), so you'll have it in your hands very soon!

Awsome @Eilon! I, must admit I am stalking your github activity for a while already 😉, and saw you checked in Blazor WebView for Android already! Will it come for IOS and Android too? Maybe intended for Preview 4 and 5 already? No worries - we tell no one 😋

**Comment by realZhangChi** [comment]
Hi, @Eilon 
I see that the [6.0-preview4](https://github.com/dotnet/aspnetcore/milestone/145) milestone has been closed but not yet released. Will blazor desktop be released in the last few days along with preview4 or will it wait until [6.0-preview5](https://github.com/dotnet/aspnetcore/milestone/146)?😊

**Comment by Eilon** [comment]
It's in preview 4 but coming from the MAUI repo so the bug tracking is a bit weird. It's a bit confusing even for me!

**Comment by realivanjx** [comment]
Will blazor desktop have something like [this](https://stackoverflow.com/questions/35660043/how-to-customize-the-window-title-bar-of-an-electron-app)? It is basically just creating a frameless window and let custom div to drag and move the window. Thanks.

**Comment by danroth27** [comment]
@realivanjx That pattern I believe should be possible, but you can also compose with native UI in a .NET MAUI Blazor app, so you might be able to implement the tabs natively, and then have a BlazorWebView per tab. This is hypothetical as I haven't actually tried it.

**Comment by realivanjx** [comment]
Is blazor desktop on winforms and wpf supported by MS or is this thing is exclusively for maui and winui? Be able to run this back to win7 sp1 will be a huge advantage at least for me. Thanks.

**Comment by danroth27** [comment]
We are providing BlazorWebView controls for WinForms and WPF as well: https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-3/#blazorwebview-controls-for-wpf-windows-forms

**Comment by realivanjx** [comment]
Also one more thing. I have installed ms edge chromium latest version but my app still asked me to install webview2 component separately. Is this expected?

**Comment by danroth27** [comment]
@realivanjx Yes, WebView2 isn't distributed with the stable Edge channel, although it is with the insider builds.

**Comment by realivanjx** [comment]
Is there any way to open dev tool?

**Comment by realZhangChi** [comment]
> Is there any way to open dev tool?

Press ctrl  + shift + i, try this.

**Comment by Xyncgas** [comment]
> Trying to run web apps on desktop is plain stupid. If you want desktop app, build a native app, so simple.
> You should focus your energy to make Blazor to work better on browsers.

it's unifying technology, you autistic freak, why are you rejecting its beauty

**Comment by Xyncgas** [comment]
ahh the eternal September is here for us all

**Comment by Xyncgas** [comment]
Although you have been triggering people with all the right spots I wonder if you are trolling. But, the things you mentioned, if that's it then yea you are pretty stupid because apparently you aren't that educated on programming with computers.

**Comment by Pilchie** [comment]
@Xyncgas - Keep comments on topic, and avoid personal attacks.

**Comment by Xyncgas** [comment]
sorry

**Comment by realivanjx** [comment]
is there any guide on how to access the content inside wwwroot folder? i tried to use httpclient but it gives me empty base uri error. tried to set the base to 0.0.0.0 but it says the address is not a routable address. thanks.

**Comment by nvmkpk** [comment]
> is there any guide on how to access the content inside wwwroot folder? i tried to use httpclient but it gives me empty base uri error. tried to set the base to 0.0.0.0 but it says the address is not a routable address. thanks.

I think you would need to use regular `System.IO.File` to read the files.

**Comment by Eilon** [comment]
@realivanjx it depends on where you're trying to read it from. If you're just trying to use it within the web view (e.g. for CSS, images, etc.) you can reference it directly from HTML content, for example, `<img src="content/flower.jpg" />`. If it's from elsewhere in the app, it depends on what you're trying to do, but you usually wouldn't read files from the `wwwroot` folder from elsewhere in the app.

**Comment by Xyncgas** [comment]
@realivanjx If you are hosting your web on a server, you can always access it with relative or absolute (https://(your server's domain)/(your file)) url, within the app, adding '/' first means you are navigating from the root

**Comment by realivanjx** [comment]
yes it works if i use it directly on the html or razor components such as img etc. what im looking at is like the weather forecast example but running it with blazor desktop. something like this in the default project template's `FetchData.razor`:
```
@code {
    private WeatherForecast[] forecasts;

    protected override async Task OnInitializedAsync()
    {
        forecasts = await Http.GetFromJsonAsync<WeatherForecast[]>("sample-data/weather.json");
    }

    public class WeatherForecast
    {
        public DateTime Date { get; set; }

        public int TemperatureC { get; set; }

        public string Summary { get; set; }

        public int TemperatureF => 32 + (int)(TemperatureC / 0.5556);
    }
}
```

**Comment by Eilon** [comment]
@realivanjx in that case I would read the file directly. In Preview 5 it's a bit awkward to read it directly and that's something we aim to fix in an upcoming milestone using a feature called MauiAssets. You'll be able to use a simple API to read these files directly without trying to use HTTP (which doesn't really make sense in a desktop app).

Another thing to keep in mind is to make this more platform-independent (such as Blazor Desktop and also Blazor WebAssembly and Blazor Server), you might want to introduce a custom service in the Dependency Injection (DI) container, such as IForecastLoader, for which you have a different implementation on each platform. In Blazor Desktop it could use MauiAssets, on Blazor WebAssembly it could use HttpClient, and on Blazor Server maybe HttpClient or something else.

**Comment by Webreaper** [comment]
For Blazor server, presumably, the best option would be to use [UseStaticFiles](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.staticfileextensions.usestaticfiles?view=aspnetcore-5.0)? I use this in my project and it works very well. 

            app.UseStaticFiles(new StaticFileOptions
            {
                FileProvider = new PhysicalFileProvider("/path/to/my/files"),
                RequestPath = "/forecasts"
            });

You'd then call something like 

           forecasts = await Http.GetFromJsonAsync<WeatherForecast[]>("/forecasts/weather.json");

I haven't tried Maui or Blazor desktop yet, so can't comment on that side of things.

**Comment by shashank-shekhar** [comment]
I don't see this in VS 2020 17.0.0 Preview 2 - has this feature not been shipped yet or am I missing something ?

**Comment by danroth27** [comment]
Hi @shashank-shekhar. This work is happening as part of .NET 6. The .NET MAUI Blazor app templates don't show up in VS2022 yet, but they will in an upcoming VS2022 preview update.

**Comment by realivanjx** [comment]
is there any way to hide or package the wwwroot files inside the app's exe or app data folder so the end user wont easily modify those files? thanks.

**Comment by Eilon** [comment]
@realivanjx that's not something that we have planned in the initial release.

**Comment by Webreaper** [comment]
Presumably if this was really important, you could embed the files in blob or (encrypted) zip file, and then use a controller as a facade in front of those files, which would extract the files on-demand as required at runtime. Messy, but would probably work?

**Comment by Eilon** [comment]
The question that comes up when people ask about embedding files is: what are you trying to protect? Ultimately there's a WebView in there that people can "sniff" into in various ways. There should be no genuine security concerns because there is no trust at that level. Users can't modify application files if they're not admins, and if they are admins, or it's locally installed, they can see/modify whatever they want and you can't prevent that no matter how hard you try. I think some developers want to prevent users from _casually_ seeing or modifying app files (that is, without much effort). My personal view is that that's somewhat pointless because why bother, but certainly others have a different view on that.

We might at some point offer ways to make it easier to embed the files, but to be clear, it offers no additional security in a trust boundary. It only adds obscurity/obfuscation.

**Comment by realivanjx** [comment]
in my case i dont really seek security that much. this is just matter of tidiness since the app is portable one. for now i managed to make everything into single exe file but the wwwroot folder.

**Comment by tub5** [comment]
@realivanjx can't you wrap it in an installer?

**Comment by IvanJosipovic** [comment]
I have also requested this feature in the MBB repo, https://github.com/dotnet/MobileBlazorBindings/issues/332
I think this is a highly valuable feature and conforms to how other frameworks like Electron work.

I prefer not to use installers as they are complicated and not needed for single file executables. Many modern projects simply ship a single exe per platform.

**Comment by Webreaper** [comment]
It sounds like what you're referring to is a properly compiled single-file executable for .Net apps, which doesn't self-extract to a temp location at startup but is actually a single compiled binary - ideally with the wwwroot etc type non-binary files embedded as some sort of resource. My understanding is that there's a separate issue in the .Net runtime to try and achieve this (I've got it bookmarked somewhere but can't find the link...).

**Comment by nvmkpk** [comment]
The current implementation does not let us configure the file provider it uses. I filed the issue https://github.com/dotnet/maui/issues/1349 but there hasn't been much activity on it so far.

**Comment by mikevel** [comment]
@Eilon what's the status of Linux desktop support for Blazor hybrid apps? I understand that MAUI is used to some extent, and MAUI does not support Linux.

**Comment by realZhangChi** [comment]
> @Eilon what's the status of Linux desktop support for Blazor hybrid apps? I understand that MAUI is used to some extent, and MAUI does not support Linux.

I found a related repository: https://github.com/jsuarezruiz/maui-linux

**Comment by Eilon** [comment]
Linux support is not in the plans for the current release, but it's certainly something we're thinking about. There have been a few issues logged on this before: https://github.com/dotnet/maui/issues?q=label%3A%22platform%2Flinux+%F0%9F%90%A7%22+

**Comment by aleksey-hoffman** [comment]
Guys, I'm new to .NET, how do you create a Blazor Desktop project?
I wanted to try migrating my Electron app to Blazor Desktop, but I cannot even find the docs for Blazor Dekstop.

Is it not out yet?

**dotnet**: 6.0.100-preview.7.21379.14
**Visual Studio**: Community 2022 Preview (64-bit), Version 17.0.0 Preview 3.1

**Comment by realivanjx** [comment]
Is there a way to set cookie on the webview? I need to show image to `img` tag from authenticated API. I tried invoking `fetch` with js interop and the server replies with this cookie:

`set-cookie: token=bWcnPlI4tXRKePtLOzpvsa0fCrTZ4pMAmpeCGZO30wnXo0c7157Nz3mAf7lhRs9E; expires=Wed, 13 Oct 2021 12:14:08 GMT; path=/; secure; samesite=none; httponly`

But the cookie is still not set so i cant show the image:

![image](https://user-images.githubusercontent.com/12814796/133081625-6a38f614-7fd2-48ac-b26c-173fc9edcad8.png)

Thanks.

**Comment by realivanjx** [comment]
Hi, im trying to set cookie on my webview to show image to `img` tag from authenticated API. I tried to make a `fetch` call to my authentication API and successfully got this response back:

![image](https://user-images.githubusercontent.com/12814796/133083588-ddd1346c-7159-43d7-b657-827b7592f6d3.png)

The problem is the cookie seems not to be set so the image API calls are still failing:

![image](https://user-images.githubusercontent.com/12814796/133083797-95da5e31-88d4-46fc-9bf7-d8baa538f891.png)

![image](https://user-images.githubusercontent.com/12814796/133083917-50970a76-bdcd-4b23-991e-35071a7f4d9c.png)

Any help is appreciated. Thanks.
